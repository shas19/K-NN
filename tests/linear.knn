let X = initTensor(2) in 
(let W = initTensor(3, 2) in
let X[0] = 4.0 in
(let (X[1]) = 3.0 in 
(let W[0, 0] = 1.0 in
(let W[0, 1] = -1.0 in
(let W[1, 0] = 2.0 in
(let W[1, 1] = 3.0 in
(let W[2, 0] = -1.0 in
(let W[2, 1] = 2.0 in
let Y = linear(W, X) in 
Y[0]))))))))