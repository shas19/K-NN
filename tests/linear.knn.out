<T>
  <k>
    0.10000000000000000e1 ~> .
  </k>
  <env>
    $1 |-> 16
    W |-> 3
    X |-> 0
    Y |-> 19
  </env>
  <store>
    0 |-> tensor ( 1 , 2 )
    1 |-> 0.40000000000000000e1
    2 |-> 0.30000000000000000e1
    3 |-> tensor ( 4 , 3 )
    4 |-> tensor ( 8 , 2 )
    5 |-> tensor ( 11 , 2 )
    6 |-> tensor ( 14 , 2 )
    7 |-> tensor ( 17 , 2 )
    8 |-> 0.10000000000000000e1
    9 |-> -0.10000000000000000e1
    10 |-> tensor ( 11 , 2 )
    11 |-> 0.20000000000000000e1
    12 |-> 0.30000000000000000e1
    13 |-> tensor ( 14 , 2 )
    14 |-> -0.10000000000000000e1
    15 |-> 0.20000000000000000e1
    16 |-> tensor ( 17 , 2 )
    17 |-> 0.00000000000000000e0
    18 |-> 0.00000000000000000e0
    19 |-> tensor ( 20 , 3 )
    20 |-> 0.10000000000000000e1
    21 |-> 0.17000000000000000e2
    22 |-> 0.20000000000000000e1
  </store>
  <nextLoc>
    23
  </nextLoc>
  <shape>
    W |-> 3 , 2
    X |-> 2
  </shape>
</T>
